code/modules/projectiles/guns/energy.dm

```diff
@@ -293,6 +293,250 @@
+
+	if(drop_override || !user) //If we want to drop it on the ground or there's no user.
+		cell.loc = get_turf(src) //Drop it on the ground.
+	else
+		user.put_in_hands(cell)
+
+	playsound(user, unload_sound, 25, 1, 5)
+	user.visible_message("<span class='notice'>[user] unloads [cell] from [src].</span>",
+	"<span class='notice'>You unload [cell] from [src].</span>", null, 4)
+	cell.update_icon()
+	cell = null
+
+	update_icon(user)
+
+	return TRUE
+
+//-------------------------------------------------------
+//M43 Sunfury Lasgun MK1
+
+/obj/item/weapon/gun/energy/lasgun/standard_lasrifle
+	name = "\improper TX-73 Lasrifle"
+	desc = "The TX-73 Lasrifle is the current standard issue laser rifle used by TGMC, it's known for it's multi-lens system to swap firemodes on the fly and good accuracy. It's rather heavy. Uses specialized energy cells."
+	force = 20 //Large and hefty! Includes stock bonus.
+	icon_state = "tx73"
+	item_state = "tx73"
+	max_shots = 50 //codex stuff
+	load_method = CELL //codex stuff
+	ammo = /datum/ammo/energy/lasgun/standard_lasrifle
+	ammo_diff = null
+	cell_type = null
+	charge_cost = TX73_STANDARD_AMMO_COST
+	attachable_allowed = list(
+						/obj/item/attachable/bayonet,
+						/obj/item/attachable/reddot,
+						/obj/item/attachable/verticalgrip,
+						/obj/item/attachable/angledgrip,
+						/obj/item/attachable/lasersight,
+						/obj/item/attachable/gyro,
+						/obj/item/attachable/flashlight,
+						/obj/item/attachable/bipod,
+						/obj/item/attachable/magnetic_harness,
+						/obj/item/attachable/attached_gun/grenade,
+						/obj/item/attachable/scope,
+						/obj/item/attachable/attached_gun/flamer,
+						/obj/item/attachable/attached_gun/shotgun,
+						/obj/item/attachable/scope/mini)
+		overcharge_position = 1
+	overcharge_datums = list(
+						/datum/lasgun/base/standard,
+						/datum/lasgun/base/overcharge,
+						/datum/lasgun/base/heat,
+						/datum/lasgun/base/wide,
+						/datum/lasgun/base/focus,
+						/datum/lasgun/base/pulse)
+
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_CAN_POINTBLANK|GUN_AMMO_COUNTER|GUN_ENERGY|GUN_AMMO_COUNTER
+	starting_attachment_types = list(/obj/item/attachable/attached_gun/grenade, /obj/item/attachable/stock/tx73)
+	attachable_offset = list("muzzle_x" = 32, "muzzle_y" = 18,"rail_x" = 12, "rail_y" = 23, "under_x" = 23, "under_y" = 15, "stock_x" = 22, "stock_y" = 12)
+
+	accuracy_mult_unwielded = 0.5 //Heavy and unwieldy; you don't one hand this.
+	scatter_unwielded = 100 //Heavy and unwieldy; you don't one hand this.
+	damage_falloff_mult = 0.25
+	fire_delay = 3
+
+//This is where you balance the Lasgun.
+/datum/lasgun/base
+	var/charge_cost = 0
+	var/ammo = null
+	var/fire_delay = 0
+	var/fire_sound = null
+	var/message_to_user = ""
+
+/datum/lasgun/base/standard
+	charge_cost = 10
+	ammo = /datum/ammo/energy/lasgun/M43
+	fire_delay = 3
+	fire_sound = 'sound/weapons/guns/fire/laser3.ogg'
+	message_to_user = "You set the Lasgun's charge mode to standard fire."
+
```

Comment:
Then for each of these you would add `icon_state = "tx73_whateverYouWant"`
as well as the relevantly named icon sprites in the .dmi